File: FilterPalindromeWords.py
# Filter Palindrome Words
# - Given a list of words, return only palindrome words.
# - Time: O(n * m), Space: O(n)
#   where m = average length of words

class Solution:
    def filterPalindromes(self, words: list[str]) -> list[str]:
        return [w for w in words if w.lower() == w.lower()[::-1]]

File: FilterPalindromeWords.cs
// Filter Palindrome Words
// - Given a list of words, return only palindrome words.

using System;
using System.Collections.Generic;
using System.Linq;

class Solution
{
    public List<string> FilterPalindromes(List<string> words)
    {
        return words.Where(w => w.ToLower() == new string(w.ToLower().Reverse().ToArray())).ToList();
    }
}
